protocols/delegate-factory/contracts/DelegateFactory.sol

```diff
@@ -17,22 +17,34 @@
 pragma solidity 0.5.10;
 
 import "@airswap/delegate/contracts/Delegate.sol";
+import "@airswap/swap/contracts/interfaces/ISwap.sol";
 import "@airswap/indexer/contracts/interfaces/ILocatorWhitelist.sol";
 import "@airswap/delegate-factory/contracts/interfaces/IDelegateFactory.sol";
 
 contract DelegateFactory is IDelegateFactory, ILocatorWhitelist {
 
   mapping(address => bool) internal deployedAddresses;
+  ISwap public swapContract;
 
   /**
-    * @notice Create a new Delegate contract
+    * @dev swapContract is unable to be changed after the factory sets it
     * @param _swapContract address of the swap contract the delegate will deploy with
+    */
+  constructor(ISwap _swapContract) public {
+    // Ensure a swap contract is provided.
+    require(address(_swapContract) != address(0),
```

Comment:
if you use ISwap then this part can never happen.

Reply:
If I remove this check one could still construct a `DelegateFactory` that constructs `Delegates` pointed to 0x0 address no?

```let delegateFactory = await DelegateFactory.new(swapContract)```